ocs_ci/deployment/ibm.py

```diff
@@ -0,0 +1,548 @@
+            in node['metadata']['labels']
+        ]
+        worker_node_names = [node['metadata']['name'] for node in worker_nodes]
+        logger.warning(worker_node_names)
+
+        # for worker_node in worker_node_names:
+        #     rookstring = f'oc debug node/'+ worker_node +
+        #                   ' -- chroot /host rm -rfv /var/lib/rook'
+        #     out = run_cmd(rookstring)
+        #     logger.warning(out)
+
+        #     for devicepath in alldevicepaths:
+        #          sgdiskstring = f'oc debug node/' + worker_node +
+        #                          ' -- chroot /host sgdisk --zap-all ' + deviepath
+        #          out = run_cmd(sgdiskstring)
+        #          logger.warning(out)
+
+        # STEP 8
+        for sc in scandlvlist:
+            lvdelete = 'oc delete localvolume -n local-storage --wait=true ' + sc[0]
+            logger.warning(lvdelete)
+            # run_cmd(lvdelete)
+            pvdelete = 'oc delete pv -l \
+                storage.openshift.com/local-volume-owner-name=' + sc[0] \
+                + ' --wait --timeout=5m'
+            logger.warning(pvdelete)
+            # run_cmd(pvdelete)
+            scdelete = 'oc delete storageclass ' + sc[1] + ' --wait=true --timeout=5m'
+            logger.warning(scdelete)
+            # run_cmd(scdelete)
+
+            # for worker_node in worker_node_names:
+            #     rookstring = 'oc debug node/'+ worker_node +
+            #                  ' -- chroot /host rm -rfv /mnt/local-storage/ ' + sc[1]
+            #     out = run_cmd(rookstring)
+            #     logger.warning(out)
+
+        # STEP 10
+        for sc in scopenshiftprovisioner:
+            scdelete = 'oc delete storageclass ' + sc + ' --wait=true --timeout=5m'
+            logger.warning(scdelete)
+            # run_cmd(scdelete)
+
+        # STEP 11
+        # Unlabel nodes
+        unlabel1 = 'oc label nodes  --all cluster.ocs.openshift.io/openshift-storage-'
+        logger.warning(unlabel1)
+        # run_cmd(unlabel1)
+
+        unlabel2 = ' oc label nodes  --all topology.rook.io/rack-'
+        logger.warning(unlabel2)
+        # run_cmd(unlabel2)
+
+        # STEP 12
+        # Remove CRDs
+        crds = 'oc delete crd backingstores.noobaa.io bucketclasses.noobaa.io \
+        cephblockpools.ceph.rook.io cephclusters.ceph.rook.io \
+        cephfilesystems.ceph.rook.io cephnfses.ceph.rook.io \
+        cephobjectstores.ceph.rook.io cephobjectstoreusers.ceph.rook.io \
+        noobaas.noobaa.io ocsinitializations.ocs.openshift.io \
+        storageclusterinitializations.ocs.openshift.io \
+        storageclusters.ocs.openshift.io  --wait=true --timeout=5m'
+        logger.warning(crds)
+        # run_cmd(crds)
+
+    def deploy_ocs(self):
+        """
+        Handle OCS deployment, since OCS deployment steps are common to any
+        platform, implementing OCS deployment here in base class.
```

Comment:
Like this docstring suggests, OCS deployment should be common regardless of platform so the deployment is implemented in the base class. If there IS something that needs to change for a particular platform, we should have some conditional code in the base class that handles it. We shouldn't overriding the `deploy_ocs` method with mostly the same code.

Reply:
I duplicated deploy_ocs due to the code related to the validation of noobaa-endpoints. We need to support noobaa so this will get cleaned up in future.

Reply:
The only thing that changed is the commented out code. If this code doesn't need to execute for IBM power you simply need to use an `if` to make sure it doesn't. We don't need to duplicate the entire function to change this behavior.

Reply:
Done. Removed redundant code from ibm.py and moved IBM Power specific code to deployment.py with conditions.